spring-cloud-stream/src/main/java/org/springframework/cloud/stream/function/StreamBridgeUtils.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2020-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.function;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.function.Function;
+
+import org.springframework.beans.factory.SmartInitializingSingleton;
+import org.springframework.cloud.function.context.FunctionCatalog;
+import org.springframework.cloud.function.context.FunctionRegistration;
+import org.springframework.cloud.function.context.FunctionRegistry;
+import org.springframework.cloud.function.context.FunctionType;
+import org.springframework.cloud.function.context.catalog.BeanFactoryAwareFunctionRegistry.FunctionInvocationWrapper;
+import org.springframework.cloud.stream.binder.ProducerProperties;
+import org.springframework.cloud.stream.config.BindingProperties;
+import org.springframework.cloud.stream.config.BindingServiceProperties;
+import org.springframework.cloud.stream.messaging.DirectWithAttributesChannel;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+import org.springframework.util.MimeType;
+import org.springframework.util.MimeTypeUtils;
+
+/**
+ * A class which allows user to send data to an output binding.
+ * While in a common scenario of a typical spring-cloud-stream application user rarely
+ * has to manually send data, there are times when the sources of data are outside of
+ * spring-cloud-stream context and therefore we need to bridge such foreign sources
+ * with spring-cloud-stream.
+ * <br><br>
+ * This utility class allows user to do just that - <i>bridge non-spring-cloud-stream applications
+ * with spring-cloud-stream</i> by providing a mechanism (bridge) to send data to an output binding while
+ * maintaining the  same invocation contract (i.e., type conversion, partitioning etc) as if it was
+ * done through a declared function.
+ *
+ * @author Oleg Zhurakousky
+ * @since 3.0.3
+ *
+ */
+public final class StreamBridgeUtils implements SmartInitializingSingleton {
```

Comment:
If we name an API class as Utils, it might give the impression that the class contains static utility methods. But we provide `StreamBridgeUtils` as part of the SCSt API and intend to be used for injection.  Can we name it to a first class `StreamBridge`?